### ISSUE TITLE
Issues with MATCH

### ISSUE THREAD

**Issue by hrstoyanov** [orientechnologies/orientdb#8414]
## Expected behavior  
I cannot get the MATCH queries to work at all for 2 vertexes (Account, Staffer) connected with an edge (WorksFor):

```
match {class: Account, as: a}-WorksFor-{class: Staffer, as: s} return s
match {class: Account, as: a} return a
match {class: Account, as: a}--{class: WorksFor, as: w}--{class: Staffer, as: s} return a,s,w
```

 See the below example:

## Steps to reproduce  
```
orientdb {db=rehabowner.db}> select from Account

+----+-----+-------+--------+----+-------------+--------+------------------+--------------------+
|#   |@RID |@CLASS |stripeId|name|emailAddress |livemode|subscriptionStatus|subscriptionLivemode|
+----+-----+-------+--------+----+-------------+--------+------------------+--------------------+
|0   |#65:0|Account|test    |Test|test@test.com|false   |active            |false               |
+----+-----+-------+--------+----+-------------+--------+------------------+--------------------+

1 item(s) found. Query executed in 0.002 sec(s).
orientdb {db=rehabowner.db}> select from Staffer

+----+-----+-------+------------------------------+--------------------+---------+--------+-------------+-------------+-------+-------+
|#   |@RID |@CLASS |auth0Id                       |email               |firstName|lastName|phoneVerified|emailVerified|blocked|created|
+----+-----+-------+------------------------------+--------------------+---------+--------+-------------+-------------+-------+-------+
|0   |#89:0|Staffer|auth0|5b3cfedea177ac1965fab185|test@test.com|Hristo   |Stoyanov|             |true         |       |#209:0 |
+----+-----+-------+------------------------------+--------------------+---------+--------+-------------+-------------+-------+-------+

1 item(s) found. Query executed in 0.001 sec(s).
orientdb {db=rehabowner.db}> select from WorksFor

+----+-----+--------+-----+-----+-----+
|#   |@RID |@CLASS  |state|in   |out  |
+----+-----+--------+-----+-----+-----+
|0   |#97:0|WorksFor|A    |#89:0|#65:0|
+----+-----+--------+-----+-----+-----+

1 item(s) found. Query executed in 0.001 sec(s).

orientdb {db=rehabowner.db}> describe WorksFor

CLASS 'WorksFor'

Records..............: 1
Super classes........: [E, Tracker]
Default cluster......: worksfor (id=97)
Supported clusters...: worksfor(97), worksfor_1(98), worksfor_2(99), worksfor_3(100), worksfor_4(101), worksfor_5(102), worksfor_6(103), worksfor_7(104)
Cluster selection....: round-robin
Oversize.............: 0.0

PROPERTIES
+----+------------+-----------------+---------+--------+--------+----+----+-------+-------+
|#   |NAME        |LINKED-TYPE/CLASS|MANDATORY|READONLY|NOT-NULL|MIN |MAX |COLLATE|DEFAULT|
+----+------------+-----------------+---------+--------+--------+----+----+-------+-------+
|0   |created     |Action           |false    |false   |false   |    |    |default|       |
|1   |lastModified|Action           |false    |false   |false   |    |    |default|       |
|2   |deactivated |Action           |false    |false   |false   |    |    |default|       |
|3   |in          |Staffer          |true     |false   |true    |    |    |default|       |
|4   |roles       |STRING           |false    |false   |false   |    |    |default|       |
|5   |state       |                 |true     |false   |false   |    |    |default|P      |
|6   |out         |Account          |true     |false   |true    |    |    |default|       |
+----+------------+-----------------+---------+--------+--------+----+----+-------+-------+

INDEXES (1 altogether)
+----+---------------+----------+
|#   |NAME           |PROPERTIES|
+----+---------------+----------+
|0   |WorksFor.out_in|[in,out]  |
+----+---------------+----------+
orientdb {db=rehabowner.db}>

orientdb {db=rehabowner.db}> match {class: Account, as: a}-WorksFor-{class: Staffer, as: s} return s

Error: com.orientechnologies.orient.core.sql.OCommandSQLParsingException: Error parsing query:
match {class: Account, as: a}
                            ^
Encountered "<EOF>" at line 1, column 29.
Was expecting one of:
    <RETURN> ...
    "," ...
    "." ...
    "<" ...
    "--" ...
    "-" ...

        DB name="rehabowner.db"
        Error Code="1"
        DB name="rehabowner.db"

!Unrecognized command: '-WorksFor-{class: Staffer, as: s}'
!Unrecognized command: 'return s'
orientdb {db=rehabowner.db}>

orientdb {db=rehabowner.db}> MATCH {Class: Account, as: a} RETURN a;

Error: com.orientechnologies.orient.core.sql.OCommandSQLParsingException: Error parsing query:
match {Class: Account, as: a}
                            ^
Encountered "<EOF>" at line 1, column 29.
Was expecting one of:
    <RETURN> ...
    "," ...
    "." ...
    "<" ...
    "--" ...
    "-" ...

        DB name="rehabowner.db"
        Error Code="1"
        DB name="rehabowner.db"

!Unrecognized command: 'RETURN a'


```

**Comment by hrstoyanov** [comment]
In addition, I am getting timeouts:
```
orientdb {db=rehabowner.db}> select out('WorksFor') from Account

Error: com.orientechnologies.common.io.OIOException: Read timed out

Error: java.net.SocketTimeoutException: Read timed out

```

**Comment by hrstoyanov** [comment]
Could this be a because I have not compiled in any of the Gremlin stuff into the server?

**Comment by luigidellaquila** [comment]
Hi @hrstoyanov 

I think the problem is in the console, please try the queries from Studio, I'm pretty sure they work well.
I'll try to fix the problem asap

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @hrstoyanov 

I just pushed a fix, it will be released with v 3.0.5

Thanks

Luigi

**Comment by hrstoyanov** [comment]
Thanks @luigidellaquila ,
Issues like this waste a lot of time (for a new ODB user) and I start to have serous doubts about the code quality and readiness of ODB and would hesitate to recommend ODB for large projects! This should have been caught by your test cases even before releasing 3.0.0!

I am completely unable to make any progress with my ODB-based project due to issues like this, as well as  #8415 , #8419.

When is 3.0.5 out?
